mmpose/datasets/datasets/top_down/topdown_mpii_dataset.py

```diff
@@ -0,0 +1,221 @@
+
+import numpy as np
+from scipy.io import loadmat, savemat
+
+from ...registry import DATASETS
+from .topdown_base_dataset import TopDownBaseDataset
+
+
+@DATASETS.register_module()
+class TopDownMpiiDataset(TopDownBaseDataset):
+    """MPII Dataset for top-down pose estimation.
+
+    The dataset loads raw features and apply specified transforms
+    to return a dict containing the image tensors and other information.
+
+    Args:
+        ann_file (str): Path to the annotation file.
+        img_prefix (str): Path to a directory where images are held.
+            Default: None.
+        data_cfg (dict): config
+        pipeline (list[dict | callable]): A sequence of data transforms.
+        test_mode (bool): Store True when building test or
+            validation dataset. Default: False.
+    """
+
+    def __init__(self,
+                 ann_file,
+                 img_prefix,
+                 data_cfg,
+                 pipeline,
+                 test_mode=False):
+        super().__init__(
+            ann_file, img_prefix, data_cfg, pipeline, test_mode=test_mode)
+
+        self.ann_file = ann_file
+        self.ann_info['flip_pairs'] = [[0, 5], [1, 4], [2, 3], [10, 15],
+                                       [11, 14], [12, 13]]
+        self.ann_info['upper_body_ids'] = (7, 8, 9, 10, 11, 12, 13, 14, 15)
+        self.ann_info['lower_body_ids'] = (0, 1, 2, 3, 4, 5, 6)
+
+        self.ann_info['use_different_joints_weight'] = False
+
+        assert self.ann_info['num_joints'] == 16
+        self.ann_info['joints_weight'] = np.ones(
+            (self.ann_info['num_joints'], 1), dtype=np.float32)
+
+        self.db = self._get_db()
+        self.image_set = set([x['image_file'] for x in self.db])
+        self.num_images = len(self.image_set)
+
+        print(f'=> num_images: {self.num_images}')
+        print(f'=> load {len(self.db)} samples')
+
+    def _get_db(self):
+        # create train/val split
+        with open(self.ann_file) as anno_file:
+            anno = json.load(anno_file)
+
+        gt_db = []
+        for a in anno:
+            image_name = a['image']
+
+            center = np.array(a['center'], dtype=np.float)
+            scale = np.array([a['scale'], a['scale']], dtype=np.float)
+
+            # Adjust center/scale slightly to avoid cropping limbs
+            if center[0] != -1:
+                center[1] = center[1] + 15 * scale[1]
+                scale = scale * 1.25
+
+            # MPII uses matlab format, index is 1-based,
+            # we should first convert to 0-based index
+            center = center - 1
+
+            joints_3d = np.zeros((self.ann_info['num_joints'], 3),
+                                 dtype=np.float)
+            joints_3d_visible = np.zeros((self.ann_info['num_joints'], 3),
+                                         dtype=np.float)
+            if not self.test_mode:
+                joints = np.array(a['joints'])
+                joints[:, 0:2] -= 1
```

Comment:
since this is an inplace operation, we need to make sure that it is modifying a copy of the annotation. i'm not sure if the np.array above line would guarantee a copy. could you double check that?

Reply:
another idea. since joints_3d is new object. how about making changes in L90 instead of L84 here?

Reply:
Yeah. It seems better to edit it in L90.